Added parse library for parsing floating point

dhylands opened the pull request.

fabricedesre commented on parse/src/lib.rs:
```diff
@@ -0,0 +1,143 @@
+        let ref_num = string.parse::<f64>();
+
+        println!("{:?} {:?}", tst_num, ref_num);
+        assert_eq!(format!("{:?}", tst_num), format!("{:?}", ref_num));
```
Why aren't you directly doing assert_eq!(tst_num, ref_num) ?

dhylands replied:
Because with floating point, the internal representation could be different, even though it prints the same. So I'd have to compare with an epsilon. All I really care about is whether its the same as far as significant digits are concerned.

Also parse::ParseFloatError is not the same struct as std::num::ParseFloatError (which is republished core::num::ParseFloatError, so I'd have to write special tests. Comparing the strings seemed easier.

fabricedesre commented on parse/src/lib.rs:
```diff
@@ -0,0 +1,143 @@
+    ParseFloatError { kind: FloatErrorKind::Invalid }
+}
+
+pub fn parse_f64(string: &str) -> Result<f64, ParseFloatError> {
```
I think you can just return a Result<f64, FloadErrorKind> (and rename FloatErrorKind to ParseFloatError)

dhylands replied:
I was going for compatability with the original parse routine. In rust, FloatErrorKind is private and ParseFloatError is the thing that can implement the Error trait, etc. I haven't done that yet, since it wasn't needed.

dhylands commented:
I updated this PR to create a crate called numeric_uitils.

I moved format.rs into the numeric_utils crate.
I added parse.rs.

I extracted the tests into separate files in the tests directory. Since each test file in the tests directory is treated as a separate crate, this allows the tests to use std and allows the lib to not need std.
